src/main/java/graphql/schema/PropertyDataFetcherHelper.java

```diff
@@ -122,22 +135,10 @@ private static Object getPropertyViaGetterMethod(Object object, String propertyN
 
     private static Object getPropertyViaGetterUsingPrefix(Object object, String propertyName, String prefix, MethodFinder methodFinder, DataFetchingEnvironment environment) throws NoSuchMethodException {
         String getterName = prefix + propertyName.substring(0, 1).toUpperCase() + propertyName.substring(1);
-        try {
-            Method method = methodFinder.apply(object.getClass(), getterName);
-            if (takesDataFetcherEnvironmentAsOnlyArgument(method)) {
```

Comment:
previously we always did this method call - now we cache this knowledge just like the method itself